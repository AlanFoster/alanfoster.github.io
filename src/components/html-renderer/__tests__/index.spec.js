import Component from "../";
import React from "react";
import renderer from "react-test-renderer";

jest.mock("../../spoilers/index.js", () => "Spoilers");

describe("html-renderer", function () {
  describe("when there is a heading", function () {
    it("does renders correctly", function () {
      const ast = {
        type: "root",
        children: [
          {
            type: "element",
            tagName: "h2",
            properties: {
              id: "heading-example",
            },
          },
        ],
      };

      const tree = renderer.create(<Component ast={ast} />);

      expect(tree).toMatchSnapshot();
    });
  });

  describe("when there are spoilers present", function () {
    it("does renders correctly", function () {
      const ast = {
        type: "root",
        children: [
          {
            type: "element",
            tagName: "spoilers",
            properties: {},
            children: [
              {
                type: "text",
                value: "Text which should be hidden",
              },
            ],
          },
        ],
      };

      const tree = renderer.create(<Component ast={ast} />);

      expect(tree).toMatchSnapshot();
    });
  });
});
