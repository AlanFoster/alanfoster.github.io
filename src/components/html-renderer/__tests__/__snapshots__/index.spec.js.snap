// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html-renderer when there are spoilers present does renders correctly 1`] = `
<div>
  <div>
    <Spoilers>
      Text which should be hidden
    </Spoilers>
  </div>
</div>
`;

exports[`html-renderer when there is a heading does renders correctly 1`] = `
<div>
  <div>
    <h2
      id="heading-example"
    />
  </div>
</div>
`;
